### PR TITLE
Live and historical dashboarding and plotting

### PR DESCRIPTION
**Pull request by LinfanS**
This pull request adds the dashboarding for the plants data using both reading from both the short term and long term storage. Code has also been refactored but would like feedback and removing redundant comments where necessary as this is quite a long script.

Contributors:
Overall dashboard design and live data filtering/plotting: Anna
Historical data filtering/plotting: Linfan

### CODE REVIEW

File: dashboard/dashboard.py [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+def get_botanists(conn: Connection) -> list[str]:
+    """
+    Obtains all possible botanists' names
+    """
+    query = """SELECT name FROM alpha.botanist"""
+    with conn.cursor() as cur:
+        cur.execute(query)
+        all_data = cur.fetchall()
+    conn.commit()
+    names = [name[0] for name in all_data]
```
Comment (HighestAuto) [inline dashboard/dashboard.py:39]: magic number

File: dashboard/get_long_term.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    """Initiates a connection to the S3 AWS Cloud using required credentials."""
+    aws_client = boto3.client(
+        "s3",
+        aws_access_key_id=getenv("ACCESS_KEY_ID"),
+        aws_secret_access_key=getenv("SECRET_ACCESS_KEY"),
+    )
+    return aws_client
+
+
+def get_file_names(client, bucket_name: str) -> list[str]:
```
Comment (MilesDrabwell) [inline dashboard/get_long_term.py]: type hint for client?

File: dashboard/get_long_term.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+def combine_transaction_data_files(filenames: list[str]) -> pd.DataFrame:
+    """Converts parquet files to a single pandas dataframe."""
+    df = pd.DataFrame()
+    for filename in filenames:
+        parquet = pd.read_parquet(filename)
+        df = pd.concat([df, parquet], ignore_index=True)
+    return df
+
+def get_long_term_data() -> pd.DataFrame:
```
Comment (MilesDrabwell) [inline dashboard/get_long_term.py:44]: Should this have a doc string?

**Review by MilesDrabwell — APPROVED** [review]
Couple clean ups but otherwise great.

File: dashboard/dashboard.py [hunk 4]
```diff
@@ -0,0 +1,500 @@
...
+            if value >=3:
+                bad_temp.append(element)
+    bad_soil = []
+    if soil_fault:
+        for element, value in soil_fault.items():
+            if value >=3:
+                bad_soil.append(element)
+    warnings(bad_temp, bad_soil)
+
+#linfan woz ere
```
Comment (HighestAuto) [inline dashboard/dashboard.py:294]: ??

File: dashboard/dashboard.py [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+def get_origin_continent(conn: Connection) -> set:
+    """
+    Obtains all possible continents' names
+    """
+    query = """SELECT continent_name FROM alpha.origin_location"""
+    with conn.cursor() as cur:
+        cur.execute(query)
+        all_data = cur.fetchall()
+    conn.commit()
+    names = sorted({name[0] for name in all_data})
```
Comment (MilesDrabwell) [inline dashboard/dashboard.py:52]: maybe names could be a clearer variable name?

**Review by MilesDrabwell — COMMENTED** [review]
Nice code

File: dashboard/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+boto3
+pandas
+python-dotenv
+altair
+streamlit
+pymssql
+scipy
+# python3 -m venv .venv
```
Comment (HighestAuto) [inline dashboard/requirements.txt]: create a python venv?

**Review by annac02 — APPROVED** [review]

**Review by HighestAuto — APPROVED** [review]

**Pull request merged**